scripts/npm.py

```diff
@@ -1,9 +1,12 @@
 import subprocess
+from pathlib import Path
 
 from hatchling.builders.hooks.plugin.interface import BuildHookInterface
 
 
 class NpmBuildHook(BuildHookInterface):
     def initialize(self, version, build_data):
-        subprocess.check_output("npm ci", shell=True)
-        subprocess.check_output("npm run build", shell=True)
+        app_js = Path(self.root, "src", "pytest_html", "resources", "app.js")
+        if not app_js.exists():
```

Comment:
wouldn't this check also trigger after having it build once?
please check if the git folder exists first

always bui when the git folder isthere, and if its missing, perhaps print a warning and only build if the target file is missing

additionally instead of check_output, running it with `subrocess.run(...,capture_output=True, check=True)`
might provide better output

Reply:
Updated @RonnyPfannschmidt 

Is that what you suggested?

Reply:
hmm... run is giving me issues...

Reply:
Whops, missing `shell=True`